### ISSUE TITLE
Do not install internal Git when using external one

### ISSUE THREAD

**Issue by xmedeko** [ForkIssues/TrackerWin#1022]
It would be nice if the internal git would not be installed (unpacked) when I have chosen to use the external Git in the settings. It would save some disk space and time on upgrade.

**Comment by DanPristupov** [comment]
This is too complicated and not really worth spending time on that.

**Comment by xmedeko** [comment]
SourceTree does not install embedded Git when the option is set to the system Git.

Do not know your installation so do not understand why it's so complicated. It seems simple - choose system git - delete internal one, choose internal Git - unpack it. On upgrade (first run) unpack when the internal Git is chosen only.

**Comment by DanPristupov** [comment]
> me: not really worth spending time on that

> you: SourceTree does not install embedded Git when the option is set to the system Git.

Do you see an irony here? :)

**Comment by xmedeko** [comment]
Nope, do not see the irony, unless you consider SourceTree as a wrong piece of software.

**Comment by nerodark** [comment]
Fork's git wrapping capabilities has to work with an officially tested git therefore that's probably why it's packed with Fork. I prefer to have complete capabilities instead of letting the user decide if he wants to use an external git and then use a feature with failures because the other git doesn't work as expected.

**Comment by xmedeko** [comment]
@nerodark Fork has a possibility to use external Git instance already. See _File - Preferences - Git - Git Instance_. This feature request is just not to install the internal Git, when I have chosen to use the external Git instance, because it's useless, then.

**Comment by gobater** [comment]
@xmedeko : SourceTree ist the biggest piece of $%#! I've seen in many time... it's becoming worst and worst with every release. That's the reason I've migrated to (and licensed) fork.

**Comment by DrunkRussianGun** [comment]
> use an external git and then use a feature with failures because the other git doesn't work as expected

One could cover Git integration with autotests and check external Git by running these tests (probably in a cloud): if tests passed, then external Git allowed to be used. Never mind, just my dreams of the perfect… 😄

**Comment by w5l** [comment]
First and foremost: I know I can set which git instance to use in the settings, and after pointing it to the system one everything works as I need it to. So nothing written here is a showstopper.

But.

I've set up and configured my system git install the way I need it for work. Multiple SSH keys injected from external authenticators, GPG or SSH commit signing, and different git settings depending on repositories. This is system wide and all my git-related tooling uses it.

Fork detects the system git install, but it decides to install and use it's own version by default, ignoring what I already set up. This local git install then includes another copy of SSH agent, GPG agent, and so on.

Now I'm one $PATH issue, click or setting away from having git actions done using some wrong settings, likely to find out only when a push fails because of credentials or a commit on repository of company A ends up being signed with a key for company B, or the wrong name, mail address, etc.

That has me a little annoyed: When a system git instance is detected, why would Fork force me another ~400MB git folder with an older git version?

I accept the "tested version" argument, it is valid. Just please acknowledge that your app might be installed by users who do more with git than just install Fork as "the one tool to manage everything git related". I don't want Fork install to meddle in my git installation, config, authentication, etc. I want to use it as a nice GUI to perform certain git actions, search commit history, whatever.

So I'd accept the installer telling me something along the lines of:

> This version of Fork was build and tested against git X.Y, your system has git A.B, do you want to install a local version?

Or even check on normal app start, as I might update the system get to an "untested/unsupported git version" during the lifetime of the Fork install. No problem.

Just please, give an option during install to say "I know what I'm doing, don't install a local git copy". Save the download and disk space.

And keep up the good work 🎉
